server/resource_server/utils/mocks/verification.go

```diff
@@ -16,6 +16,23 @@ type MockCodeGenerator struct {
 	VerificationCode string
 }
 
-func (g *MockCodeGenerator) GenerateCode(emailAddress string) string {
-	return g.VerificationCode
+func (g *MockCodeGenerator) GenerateCode(user *models.User, emailAddress string) (string, error) {
```

Comment:
It is here I wish I had more experience to give better feedback about best practices in terms of where to code and store Mocks for future use. You might be able to automatically generate a mock for the `Generator` interface in `...code/generator.go` I think this is what Firhan does.

Reply:
Yeah, we can do that. The reason why I didn't do this here is because I'd need to make significant refactoring in the existing tests to adopt the use of generated mocks instead of the current approach. I can create a separate issue and implement this within both the email verification codebase, as well as in other places.